lib/bootloader_setup.pm

```diff
@@ -545,6 +545,12 @@ sub specific_caasp_params {
 }
 
 sub tianocore_enter_menu {
+    # wait for tianocore boot screen
+    # we need to reduce this waiting time as much as possible
+    while (!check_screen('tianocore-bootscreen', 0, no_wait => 1)) {
+        sleep 0.1;
+    }
+
     # press F2 and be quick about it
     send_key_until_needlematch('tianocore-mainmenu', 'f2', 15, 1);
 }
```

Comment:
I faced the same issue yesterday as @ldevulder  (i just installed openQA for first time on aarch64). And my quick(dirty) solution was to increase count for needlematch  `send_key_until_needlematch('tianocore-mainmenu', 'f2', 90, 1);`.  Is it bad idea?  Also, isn't possible that your while loop will get stuck? (If something goes wrong).

Reply:
This code is not working at 100%, I will push a new one that works shortly that works in all my tests this time.

Increasing count is not the perfect solution, it could also failed sometimes after fencing, as you don't know the time the cluster takes to fence the node.

I'm not sure that my loop can stuck by itself, if it's the case it's because uefi fw is not booting, so all test will failed.
I have a look at a lot of `while` loop in openqa tests, and a lot does not exit the loop after a timeout or a count. But I can add it if needed, but for me I don't think so.

Reply:
If I understand well, current change emulates `send_key_until_needlematch` with unlimited count?

Reply:
yes, until we enter uefi menu. Finally like we did in real life (at least me, but I'm pretty sure to not be alone!).